devlib/platform/arm.py

```diff
@@ -103,7 +105,9 @@ def _get_target_ip_address(self, target):
             tty.sendline('su')  # this is, apprently, required to query network device
                                 # info by name on recent Juno builds...
             self.logger.debug('Waiting for the Android shell prompt.')
-            tty.expect(target.shell_prompt)
+            # In Python3 pexpect does not accept bytes-like objects, so convert to str
+            shell_prompt_string = target.shell_prompt.pattern
+            tty.expect(shell_prompt_string)
```

Comment:
Erm I'm unclear on the need for this -- `target.shell_prompt` is a regex, not a bytes-like object.

Reply:
Yes, it may not be a bytes-like object, but it is not a string, and the Python3 version of tty.expect requires a string input, so we need to extract the string using the pattern property.

Reply:
According to the docs: 

http://pexpect.readthedocs.io/en/stable/api/pexpect.html#pexpect.spawn.expect

the pattern argument could be a compiled re.

Reply:
For some reason, it doesn't work and gives:
TypeError(cannot use a string pattern on a bytes-like object)